qiskit/providers/aer/noise/device/models.py

```diff
@@ -21,34 +21,55 @@
 from numpy import inf, exp, allclose
 
 import qiskit.quantum_info as qi
-from .parameters import readout_error_values
+from .parameters import _NANOSECOND_UNITS
 from .parameters import gate_param_values
+from .parameters import readout_error_values
 from .parameters import thermal_relaxation_values
-from .parameters import _NANOSECOND_UNITS
-
 from ..errors.readout_error import ReadoutError
 from ..errors.standard_errors import depolarizing_error
 from ..errors.standard_errors import thermal_relaxation_error
+from ..noiseerror import NoiseError
 
 logger = logging.getLogger(__name__)
 
 
-def basic_device_readout_errors(properties):
+def basic_device_readout_errors(properties=None, target=None):
     """
-    Return readout error parameters from a devices BackendProperties.
+    Return readout error parameters from either of device Target or BackendProperties.
+
+    If ``target`` is supplied, ``properties`` will be ignored.
 
     Args:
         properties (BackendProperties): device backend properties
+        target (Target): device backend target
 
     Returns:
         list: A list of pairs ``(qubits, ReadoutError)`` for qubits with
         non-zero readout error values.
+
+    Raises:
+        NoiseError: if neither properties nor target is supplied.
     """
     errors = []
-    for qubit, value in enumerate(readout_error_values(properties)):
-        if value is not None and not allclose(value, [0, 0]):
-            probabilities = [[1 - value[0], value[0]], [value[1], 1 - value[1]]]
-            errors.append(([qubit], ReadoutError(probabilities)))
+    if target is None:
+        if properties is None:
+            raise NoiseError("Either properties or target must be supplied.")
+        # create from BackendProperties
+        for qubit, value in enumerate(readout_error_values(properties)):
+            if value is not None and not allclose(value, [0, 0]):
+                probabilities = [[1 - value[0], value[0]], [value[1], 1 - value[1]]]
+                errors.append(([qubit], ReadoutError(probabilities)))
+    else:
+        # create from Target
+        for q in range(target.num_qubits):
+            prop = target['measure'][(q,)]
```

Comment:
I'm not sure we want to do this with a get item here without checking that `measure` is defined in the target. I think in practical terms almost every target will have this defined but it's not guaranteed.